src/server/object_services/object_server.js

```diff
@@ -406,9 +406,17 @@ async function complete_object_upload(req) {
         set_updates.sha256_b64 = req.rpc_params.sha256_b64;
     }
 
-    const map_res = req.rpc_params.multiparts ?
-        await _complete_object_multiparts(obj, req.rpc_params.multiparts) :
-        await _complete_object_parts(obj);
+    let map_res;
+
+    if (req.rpc_params.multiparts) {
+        map_res = await _complete_object_multiparts(obj, req.rpc_params.multiparts);
+    } else if (req?.params?.num_parts === 1) {
+        //skip db access for small objects
+        dbg.log1("skipped _complete_object_multiparts");
+        map_res = {size: req.rpc_params.size, num_parts: 1};
+    } else {
+        map_res = await _complete_object_parts(obj);
+    }
```

Comment:
I would prefer to move this optimization into `_complete_object_parts()` (need to send params.num_parts inside).

Over there we can start by identifying a single part case, and optimize the hell of it, so that we don't even read the part from the db, which is an aggressive optimization which we also need to explain why it is safe (the reason is that mpu cannot have a single part per multipart... which is a bit of a "bad" assumption to take because it could easily get broken one day by someone who just doesn't expect it calling these apis).

We can also add a simpler condition that will still read the parts, but will skip updating them by noticing that they are already committed and do not require an offset/seq update - simply surround the push to parts_updates with a condition like 
```js
if (part.uncommitted || updates.set_updates) {
    context.parts_updates.push(updates);
}
```

on this line - 
https://github.com/noobaa/noobaa-core/blob/cdd266f64904a67b985fd07c9d97abfe85f43790/src/server/object_services/object_server.js#L2099

Reply:
I've moved the check whether to skip the db into _complete_next_parts().
Since we're going for performance, I figured it's best to skip the db network hop altogether.
You think it's not safe enough?